IssueCreator/IssueCreator/IssueManager.cs

```diff
@@ -0,0 +1,270 @@
+            }
+
+            // Convert to Epic
+            if (issueToCreate.CreateAsEpic)
+            {
+                await ConvertToEpicAsync(repoFromGH.Id, createdIssue.Number);
+            }
+
+            Process.Start(createdIssue.HtmlUrl);
+
+            return (true, string.Empty);
+        }
+
+        public async Task<Issue> GetIssueAsync(long repoId, int issueNumber)
+        {
+            Issue issue = await GetValueFromCache($"issue_{repoId}_{issueNumber}", () => _githubClient.Issue.Get(repoId, issueNumber));
+            return issue;
+        }
+
+        public async Task<List<IssueDescription>> GetEpicsAsync(List<string> repositoriesToUse)
+        {
+            ConcurrentBag<IssueDescription> results = new ConcurrentBag<IssueDescription>();
+
+            // defensive copy in case the list of repos changes.
+            List<string> repositories = new List<string>(repositoriesToUse);
+            ConcurrentBag<IssueDescription> result = new ConcurrentBag<IssueDescription>();
+
+            Task[] tasks = new Task[repositories.Count];
+
+            int count = 0;
+            foreach (var orgAndRepoName in repositories)
+            {
+                // for each repo, get the epics.
+                (string org, string repo) = GetOwnerAndRepoFromString(orgAndRepoName);
+
+                tasks[count++] = Task.Run(async () =>
+                {
+                    Repository gitHubRepoObj = await GetRepositoryAsync(org, repo);
+
+                    var epicList = await GetValueFromCache($"epic_{org}_{repo}", () => _zenHubClient.GetRepositoryClient(gitHubRepoObj.Id).GetEpicsAsync(), DateTimeOffset.Now.AddHours(1));
+
+                    foreach (var epic in epicList.Value.Epics)
+                    {
+                        long repoId = epic.RepositoryId;
+                        int issueNumber = epic.IssueNumber;
+                        // from the issue link, get the cached issue from the repo
+                        Issue issue = await GetIssueAsync(repoId, issueNumber);
+
+                        result.Add(new IssueDescription() { Issue = issue, Repo = gitHubRepoObj });
+                    }
+                });
+            }
+
+            await Task.WhenAll(tasks);
+
+            return result.ToList();
+        }
+
+        private (string, string) GetOwnerAndRepoFromString(string input)
+        {
+            string[] parts = input.Split('\\');
+            return (parts[0], parts[1]);
+        }
+
+        private async Task<TResult> GetValueFromCache<TResult>(string key, Func<Task<TResult>> getValue, DateTimeOffset cacheDuration = default)
+        {
+            TResult itemInCache = _cache.Get<TResult>(key);
+            if (itemInCache != null)
+            {
+                return itemInCache;
+            }
+
+            // add the value to the cache.
+
+            if (cacheDuration == default)
+            {
+                cacheDuration = DateTimeOffset.Now.AddHours(8);
```

Comment:
This always scares me.  I've never hard coded a cache timeout like this and not regretted it.  That said, I tend to open the app for minutes at a time and then close it.  I doubt most people will ever hit even the hour timeouts.

Reply:
Yeah -- not very happy about this, but I don't think it makes sense to do anything more here. This value needs to be some number, otherwise we never invalidate the values we add... open to suggestions 😀.